python/cudf/cudf/io/json.py

```diff
@@ -37,16 +37,31 @@ def read_json(
             f"or a bool, or None. Got {type(dtype)}"
         )
 
-    if engine == "cudf" and not lines:
+    if engine == "cudf_experimental":
+        raise ValueError(
+            "engine='cudf_experimental' support has been removed, "
+            "use `engine='cudf'`"
+        )
+
+    if engine == "cudf_legacy":
+        # TODO: Deprecated in 23.02, please
+        # give some time until `cudf_legacy`
+        # support can be removed completely.
```

Comment:
> "give some time until `cudf_legacy` support can be removed completely

Will [our one-release deprecation policy](https://docs.rapids.ai/api/cudf/nightly/developer_guide/contributing_guide.html#deprecating-and-removing-code) be enough for this, or do you think a special exception is warranted? If more time is needed, let's try to indicate the release in which support should be removed in the error message or perhaps a code comment.

Reply:
Yeah, we definitely want to give this a special exception. I don't think we have a specific release decided yet to remove it completely. I updated the comment here.

cc: @GregoryKimball @vuule too.

Reply:
I think we'll pick the removal release based on 23.02 feedback. In general, we'll remove the old reader as soon as there are no user issues specific to the new one.

Reply:
That sounds reasonable! I just wanted to make sure we discussed this topic before merging. 👍